helm/hello-world/values.yaml

```diff
@@ -1,111 +1,555 @@
+  # @schema
   port: 80
+
+  # @schema
+  # title: Optional service labels
+  # type: object
+  # patternProperties:
+  #   "^[a-zA-Z0-9/-_\\.]+$":
+  #     $ref: https://schema.giantswarm.io/labelvalue/v0.0.1
+  # @schema
   labels: {}
+
+  # @schema
+  # title: Optional service annotations
+  # type: object
+  # patternProperties:
+  #   "^[a-zA-Z0-9/-_\\.]+$":
+  #     type: string
+  #     title: Annotation value
+  # @schema
   annotations: {}
 
+# @schema
+# title: Ingress
+# type: object
+# @schema
 ingress:
+
+  # @schema
+  # title: Enabled
+  # type: boolean
+  # @schema
+  # If set to false, no Ingress resource will be created.
   enabled: true
+
+  # @schema
+  # title: Class
+  # type: string
+  # @schema
   className: nginx
+
+  # @schema
+  # title: Optional ingress annotations
+  # type: object
+  # patternProperties:
+  #   "^[a-zA-Z0-9/-_\\.]+$":
+  #     type: string
+  #     title: Annotation value
+  # examples:
+  #   - "kubernetes.io/ingress.class: nginx"
+  #   - "kubernetes.io/tls-acme: 'true'"
+  # @schema
   annotations:
-    # kubernetes.io/ingress.class: nginx
-    # kubernetes.io/tls-acme: "true"
+
+    # @schema
+    # required: false
+    # @schema
     cert-manager.io/cluster-issuer: letsencrypt-giantswarm
+
+  # @schema
+  # title: Hosts
+  # type: array
+  # required: true
+  # items:
+  #   type: object
+  #   properties:
+  #     host:
+  #       type: string
+  #       format: hostname
+  #       description: Hostname to expose the service for.
+  #     paths:
+  #       type: array
+  #       items:
+  #         type: object
+  #         properties:
+  #           path:
+  #             type: string
+  #           pathType:
+  #             type: string
+  #             enum:
+  #             - Prefix
+  #             - Exact
+  # @schema
+  # You will have to overwrite the host value to make it match your cluster's ingress subdomain.
   hosts:
   - host: hello.cluster.provider.gigantic.io
     paths:
     - path: /
       pathType: Prefix
+
+  # @schema
+  # title: Transport layer security (TLS)
+  # type: array
+  # items:
+  #   type: object
+  #   properties:
+  #     secretName:
+  #       type: string
+  #       description: Name of the secret containing the TLS server certificate.
+  #     hosts:
+  #       type: array
+  #       items:
+  #         type: string
+  #         format: hostname
+  #       description: Hostnames to use the TLS server certificate for.
+  # @schema
+  # You will have to overwrite the hosts values to make it match your cluster's ingress subdomain, matching the names used as ingress hosts.
   tls:
   - secretName: hello-world-tls
     hosts:
     - hello.cluster.provider.gigantic.io
 
+# @schema
+# title: Resource requests and limits
+# type: object
+# @schema
+# This is directly injected as the main pods's resource requests and limits.
 resources:
+
+  # @schema
+  # title: Requests
+  # type: object
+  # @schema
   requests:
-    cpu: 200m
-    memory: 50Mi
```

Comment:
Basically the same as for the max replicas: Those values were tuned with load tests. IIRC a single hello world pod takes this amount of resources to handle 1.000 requests per second at 100 connections or something like that. So this ratio is kind of tested. So in the end memory consumption will probably never be a multiple of CPU consumption.